tests/karate/src/test/java/fe/net/MockBackend.java

```diff
@@ -0,0 +1,102 @@
+package fe.net;
+
+import com.intuit.karate.Logger;
+import com.intuit.karate.http.WebSocketServerBase;
+
+import common.net.MessageBuffer;
+import common.net.MessageQueue;
+import karate.io.netty.channel.Channel;
+import karate.io.netty.channel.ChannelHandlerContext;
+import karate.io.netty.channel.SimpleChannelInboundHandler;
+import karate.io.netty.handler.codec.http.websocketx.TextWebSocketFrame;
+
+import java.util.concurrent.*;
+import java.util.function.Function;
+
+/** Defines a mock backend server that is fully customisable. */
+public class MockBackend extends SimpleChannelInboundHandler<TextWebSocketFrame> {
+
+  private final Logger logger = new Logger(getClass().getSimpleName());
+
+  private final MessageQueue queue;
+  private final WebSocketServerBase server;
+  // Will be set to true once the connection is established
+  private final CompletableFuture<Boolean> connected = new CompletableFuture<>();
+
+  // Defines the rule to apply on incoming messages to produce its reply.
+  // Can be null if no reply should be sent back.
+  private Function<String, String> replyProducer = ReplyMethods.ALWAYS_VALID;
```

Comment:
Don't you think this should be initialized to null, and the tester should always set a replyProducer manual for each message he/she eants to handle.

Reply:
I like thinking of the default behavior as the one that will be used in the most common case.

And for me the most common case is to always return a Result.